e2e/react/src/react.test.ts

```diff
@@ -141,57 +141,97 @@ forEachCli('nx', () => {
       });
     }, 120000);
 
-    it('should generate app with styled-components', async () => {
+    it('should generate app with different style options', async () => {
       ensureProject();
-      const appName = uniq('app');
+
+      const styledComponentsApp = uniq('app');
 
       runCLI(
-        `generate @nrwl/react:app ${appName} --style styled-components --no-interactive`
+        `generate @nrwl/react:app ${styledComponentsApp} --style styled-components --no-interactive`
       );
 
-      await testGeneratedApp(appName, {
+      // Checking that we can enable displayName just for development.
+      updateFile(
+        `apps/${styledComponentsApp}/.babelrc`,
+        JSON.stringify(
+          {
+            presets: ['@nrwl/react/babel'],
+            env: {
+              development: {
+                plugins: [
+                  [
+                    'styled-components',
+                    {
+                      pure: true,
+                      ssr: true,
+                      displayName: true,
+                    },
+                  ],
+                ],
+              },
+              production: {
+                plugins: [
+                  [
+                    'styled-components',
+                    {
+                      pure: true,
+                      ssr: true,
+                      displayName: false,
+                    },
+                  ],
+                ],
+              },
+            },
+          },
+          null,
+          2
+        )
+      );
+
+      await testGeneratedApp(styledComponentsApp, {
         checkStyles: false,
-        checkProdBuild: false,
+        checkProdBuild: true,
         checkLinter: false,
         checkE2E: false,
       });
-    }, 120000);
 
-    it('should generate app with styled-jsx', async () => {
-      ensureProject();
-      const appName = uniq('app');
+      expect(readFile(`dist/apps/${styledComponentsApp}/main.js`)).toContain(
+        'app__StyledApp'
+      );
+      expect(
+        readFile(`dist/apps/${styledComponentsApp}/main.esm.js`)
+      ).not.toContain('app__StyledApp');
+
+      const styledJsxApp = uniq('app');
```

Comment:
Why not have this be 2 separate tests?  One for styled-components, one for styledJSX?